### DISCUSSION TITLE
UR e-Series Remote Connection Tutorial

### DISCUSSION THREAD

**Discussion by madelinegannon** [visose/Robots#210] [Q&A] [answered]
A couple of ✨tips✨ for anyone working with e-Series Universal Robots — since it's slightly different than when working with earlier UR models.

- You have to be in `Remote Control` mode to receive external `.urp` scripts. Manual and Automatic did not work for me on SW 5.9.5.
  - See [How to enable Remote Control on URe](https://robodk.com/doc/en/Robots-Universal-Robots-How-enable-Remote-Control-URe.html) tutorial from RoboDK.
- [This](https://www.universal-robots.com/articles/ur/interface-communication/remote-operation-of-robots/) is a great guide for advanced remote operation of URs.

**Comment by visose** [comment, marked as answer]
Thank you @madelinegannon. Btw, this is not documented but now you can upload files from Grasshopper using the UR's ftp server. This hasn't been tested much so there could be issues, some details here about how to enable it in the Remote component:

> If you input an IP address, it will use the secondary client interface to stream the program and run the play and pause commands. You can switch to the FTP back-end by specifying the IP in the format of `sftp://ip`. This will upload the program as a .urp file to the controller using the FTP server and load it. It uses the dashboard interface for loading, play, and pause. You can change the default values for username, password, or location of the programs folder, by using: `sftp://user:pass@ip/path/to/programs`

For the ftp upload I think you won't need to enable the remote control, but you will for the play and pause buttons to work from Grasshopper.
